SWLOR.Game.Server/Service/Quest.cs

```diff
@@ -262,7 +263,11 @@ public static void ProgressKillTargetObjectives()
             var possibleQuests = GetQuestsAssociatedWithNPCGroup(npcGroupType);
             if (possibleQuests.Count <= 0) return;
 
-            var killer = GetLastKiller();
+            // We can't use GetLastKiller() as various abilities deal damage that isn't sourced from
+            // the PC.  So get the nearest PC instead.
+            // Ideally we'd use the Combat Point database but that is cleaned up in another method called
+            // by this same handler, so we can't rely on it existing.
+            var killer = GetNearestCreature(CreatureType.PlayerCharacter, 1, creature);
```

Comment:
Are we able to use the enmity service for this or does that also get cleaned up by the time this is run?

Reply:
Same race condition - it's also cleared up in the crea_death event.  If we could order the events we could use it.... is that possible?

Reply:
The order is not deterministic unfortunately. It's possible to hook the event, do everything, then fire off another script and catch it here. The handful of times I've had to do that it's worked well. I'm just worried the nearest creature might be a different PC than expected.

Reply:
But they should be in the same party at least, which will make the hook work fine. The odds of having two PCs in different parties both fighting the same creature (and caring about quest kills) don't seem high. 

I'm actually more bothered by this because GetNearestCreature is an expensive method to call and I'd rather not have to :) I was tempted to use the combat point service to tag the last PC to put a combat point on the NPC in a local variable and read that.  It would be a bastardisation of the current system which is all neatly using databases, but using a variable on the NPC means it'll go away when the NPC object is destroyed after death.

Reply:
I sent you a Discord message, but logging it here too for safekeeping: https://github.com/zunath/SWLOR_NWN/pull/1312

This PR adds "before" and "after" events to the creature events. The idea being that you could hook the "before" events and have access to all the enmity / combat point data as it doesn't get wiped until the "after" event is called. I think this should guarantee you get the right target but I have not tested it yet. Let me know when you get some time and we can look at it.

Reply:
Ok - so given the change you've made in the other PR I'll update this to pull from the CombatPoint table and pick something that did damage to the NPC.  When merged with the other PR that should 'just work'.  I won't be able to test until both are merged though,